Packages/Status/Sources/Status/Row/Subviews/StatusRowTextView.swift

```diff
@@ -1,22 +1,61 @@
 import DesignSystem
+import Env
 import Models
 import SwiftUI
 
 struct StatusRowTextView: View {
   @EnvironmentObject private var theme: Theme
+  @EnvironmentObject private var preferences: UserPreferences
 
   let status: AnyStatus
-  let viewModel: StatusRowViewModel
-
+  @ObservedObject var viewModel: StatusRowViewModel
+  
   var body: some View {
-    HStack {
-      EmojiTextApp(status.content, emojis: status.emojis, language: status.language)
-        .font(.scaledBody)
-        .emojiSize(Font.scaledBodyPointSize)
-        .environment(\.openURL, OpenURLAction { url in
-          viewModel.routerPath.handleStatus(status: status, url: url)
-        })
-      Spacer()
+    VStack {
+      HStack {
+        EmojiTextApp(status.content, emojis: status.emojis, language: status.language, lineLimit: viewModel.lineLimit)
+          .font(.scaledBody)
+          .emojiSize(Font.scaledBodyPointSize)
+          .environment(\.openURL, OpenURLAction { url in
+            viewModel.routerPath.handleStatus(status: status, url: url)
+          })
+        Spacer()
+      }
+      makeCollapseButton()
+    }
+    .onAppear {
+      viewModel.collapseLongPosts = preferences.collapseLongPosts
```

Comment:
I'll do a few modifications here. Basically this cost a lot of performances. onAppear modifying a published property (even if it's set to the same value) trigger a body calculation of everything observing the viewModel. Will probably set it only if the preference is different from the VM value.

Reply:
Is there a different way to set the initial value into the viewModel? I was not sure about acccessing the `UserPreferences` from the model although in `markSeen()` this is done as well.

Reply:
You can access it using .shared as static. It's fine if we don't observe it.